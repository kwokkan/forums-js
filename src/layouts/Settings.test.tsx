import { create } from "react-test-renderer";

test("Renders without user", () => {
    jest.isolateModules(() => {
        const mockSession = jest.fn((): [INextSession | null, boolean] => [null, true]);

        jest.doMock("next-auth/client", () => {
            return {
                __esModule: true,
                useSession: mockSession
            };
        });

        const { Settings } = require("./Settings");

        const tree = create(
            <Settings />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});

test("Renders with user", () => {
    jest.isolateModules(() => {
        const mockSession = jest.fn((): [INextSession | null, boolean] => [{
            forumsUser: {
                id: 1,
                name: "Test user",
                joinedDate: 1
            }
        }, true]);

        jest.doMock("next-auth/client", () => {
            return {
                __esModule: true,
                useSession: mockSession
            };
        });

        const { Settings } = require("./Settings");

        const tree = create(
            <Settings />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
