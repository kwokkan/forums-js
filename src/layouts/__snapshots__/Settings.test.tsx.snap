// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with user 1`] = `
<div>
  Logged in
</div>
`;

exports[`Renders without user 1`] = `
<div>
  Not logged in
</div>
`;
